### ISSUE TITLE
Shields block uploading images on GitHub

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#2034]
## Description 

Shields block uploading images on GitHub

## Steps to Reproduce

   1. Keep default shields on GitHub
   2. Create a new issue and try to upload an image via clipboard or attach files
   3. Doesn't add it. Disable Adblocker, upload works

## Actual result:

Shields block uploading images on GitHub

## Expected result:
Should not block uploading files/adding images

## Reproduces how often: 

Easy

## Brave version (brave://version info)

Brave | 0.57.6 Chromium: 71.0.3578.31 (Official Build) beta (64-bit)
-- | --
Revision | c88fdf2a4ce19a713615ca4fbde7a0d0b5fe2363-refs/branch-heads/3578@{#427}
OS | All

### Reproducible on current release:
- Does it reproduce on brave-browser dev/beta builds? 
Yes reproduces on both dev/beta

### Website problems only:  
- Does the issue resolve itself when disabling Brave Shields? 
- Is the issue reproducible on the latest version of Chrome? 

### Additional Information

Reproduced by @kjozwiak on macOS

**Comment by srirambv** [comment]
Disable and enable shields allows you to add images on creating issues

**Comment by tomlowenthal** [comment]
Couldn't repro on MacOS with version 0.57.4 [Chromium: 71.0.3578.31 (Official Build) dev]. More detailed repro steps maybe?

**Comment by Liunkae** [comment]
I can reproduce with a fresh profile. Turning off and on shields fixes the problem (seems like it's the cookie shield in particular).

Brave | 0.57.6 Chromium: 71.0.3578.31 (Official Build) beta(64-bit)
Revision | c88fdf2a4ce19a713615ca4fbde7a0d0b5fe2363-refs/branch-heads/3578@{#427}
OS | Mac OS X

1. Start with a fresh profile.
2. Go to https://github.com/brave/brave-browser/issues and login.
3. Create a new issue.
4. Try to upload an image to the issue.

![image](https://user-images.githubusercontent.com/19424103/48306969-aa08c280-e508-11e8-80e7-39367c196bc1.png)

**Comment by kjozwiak** [comment]
I couldn't reproduce this after a few attempts but I have ran into this several times. As @Liunkae mentioned, disabling/re-enabling usually fixes the issue. There's definitely an issue here where the shields gets into a state where it starts blocking uploads. Re-opening as this is still obviously and issue even though it seems to be intermittent.

**Comment by Liunkae** [comment]
+1 from #2465 with 0.57.x. I have not had this problem since updating to 0.59.x dev, so it might have been fixed.

**Comment by srirambv** [comment]
+1 from @sobolevn & @lk-geimfari via #2485

**Comment by manojmj92** [comment]
+1 on MacOS Mojave

**Comment by kjozwiak** [comment]
Ran into this again using `0.59.5 Chromium: 71.0.3578.80`... Dragging & Dropping images into GH stopped working. Disabling the shields worked.

We need to find some reliable STR but seems like it's happening intermittently.

**Comment by NejcZdovc** [comment]
+1 it's happening for me all the time `Version 0.59.2 Chromium: 71.0.3578.80 (Official Build) dev (64-bit)`

**Comment by yochem** [comment]
Happens on macOS Mojave with other sites too, not just Github.
`Version 0.57.18 Chromium: 71.0.3578.80 (Official Build) (64-bit)`.

**Comment by iefremov** [comment]
Looks like this is a dupe of https://github.com/brave/brave-browser/issues/2252
Reveting https://github.com/brave/brave-core/pull/754/files fixes it for me.

**Comment by Liunkae** [comment]
Still happening for me with Brave 0.60.3 dev on macOS 10.14. Workaround: change cookie setting to "All cookies allowed".

**Comment by mbacchi** [comment]
![github](https://user-images.githubusercontent.com/6763301/50303650-f79f1600-044a-11e9-8cc0-063ed5827e59.png)

**Comment by mbacchi** [comment]
The above is what I see in the developer console when shields are up ^^^

**Comment by HeadFox** [comment]
+1 happening on paper.dropbox too

**Comment by dy** [comment]
Yep. Getting both
```
new:1 Refused to connect to 'data:text/plain,' because it violates the following Content Security Policy directive: "connect-src 'self' uploads.github.com www.githubstatus.com collector.githubapp.com api.github.com www.google-analytics.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com wss://live.github.com".
```

```
Access to XMLHttpRequest at 'https://github-production-user-asset-6210df.s3.amazonaws.com/' from origin 'https://github.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.
```

The second is raised when copy-pasting an image into an issue input.

Turning shield off and one helps, exactly once.

**Comment by kjozwiak** [comment]
@iefremov awesome 👍

**Comment by Liunkae** [comment]
#2252 was fixed with brave/brave-core#1240, so this issue should also be fixed.

**Comment by iefremov** [comment]
Closing since #2252 should be fixed now.

**Comment by btlechowski** [comment]
Verification passed on

Brave | 0.58.20 Chromium: 71.0.3578.98 (Official Build) (64-bit)
-- | --
Revision | 15234034d19b85dcd9a03b164ae89d04145d8368-refs/branch-heads/3578@{#897}
OS | Windows 7

Image pasted successfully. Upload from a file works also.

![image](https://user-images.githubusercontent.com/34715963/50929669-1bf47e00-145e-11e9-81e9-573cd7c4b3c8.png)
